core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/RawInfraImpl.kt

```diff
@@ -511,9 +577,46 @@ class RawInfraImpl(
         route: String?,
     ): DistanceRangeMap<Speed> {
         val res = distanceRangeMapOf<Speed>()
+
+        var trainTagId = trainTag
+        // TODO remove once long names are unused
+        if (speedLimitTagNameToIdMap.contains(trainTagId)) {
+            trainTagId = speedLimitTagNameToIdMap[trainTagId]
+        }
+        val trainSpeedLimitTagDescriptor = speedLimitTagPool[trainTagId]
+
+        // TODO remove once long names are unused
+        val trainTagName = trainSpeedLimitTagDescriptor?.name
+
         for (entry in trackChunkPool[trackChunk.value].speedSections.get(trackChunk.direction)) {
             val speedSection = entry.value
-            val speedFromTag = speedSection.speedByTrainTag[trainTag]
+
+            /* SpeedLimitTag handling */
+            var infraTagSpeed = speedSection.speedByTrainTag[trainTagId]
+            // TODO remove once long names are unused
+            if (infraTagSpeed == null && trainTagName != null) {
+                infraTagSpeed = speedSection.speedByTrainTag[trainTagName]
+            }
+
+            var infraTagId = if (infraTagSpeed != null) trainTagId else null
+
+            if (infraTagSpeed == null && trainSpeedLimitTagDescriptor != null) {
+                for (fallbackTagId in trainSpeedLimitTagDescriptor.fallbackList) {
+                    val fallbackSpeed =
+                        speedSection.speedByTrainTag[fallbackTagId]
+                            // TODO remove once long names are unused
+                            ?: speedSection.speedByTrainTag[speedLimitTagPool[fallbackTagId]?.name]
+
+                    if (fallbackSpeed != null) {
+                        if (infraTagSpeed == null || fallbackSpeed > infraTagSpeed) {
+                            infraTagSpeed = fallbackSpeed
+                            infraTagId = fallbackTagId
+                        }
+                    }
+                }
+            }
```

Comment:
This feels large enough to become a separate method.
Maybe it could be a method of `SpeedSection` or of `SpeedLimitTagDescriptor`

Reply:
Method separated in https://github.com/OpenRailAssociation/osrd/pull/7903/commits/8c7694d388f024f624de3db69c97ce03a8527567#diff-5d11fb62d051b79ae994acaff5b58f193131909011f76ad3547e4c777e4dfc85R637
Although IMO it should be a method of `SpeedLimitTagDescriptor`, but currently using `speedLimitTagPool` from `RawInfraImpl`, added a [TODO](https://github.com/OpenRailAssociation/osrd/pull/7903/commits/8c7694d388f024f624de3db69c97ce03a8527567#diff-5d11fb62d051b79ae994acaff5b58f193131909011f76ad3547e4c777e4dfc85R656-R657) on that.